### PR TITLE
Reorganize some code to have 2D things grouped together

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR takes pieces out of #47054, in the same vain as how #51474 took pieces out of #50147.

* In CanvasItemEditorPlugin, move `skeleton_menu` declaration next to the rest of the skeleton code.
* In CanvasItemEditorPlugin, put `Control`-derived options before `Node2D`-derived ones.
* In SpriteFramesEditorPlugin, move the checks right after the casts they're using.
* In World2D, move get_direct_space_state method up.
* In register_scene_types.cpp, organize the 2D includes, organize 2D classes to be together, and move AudioStreamPlayer3D together with the 3D classes.
* In register_server_types.cpp, organize 2D and 3D navigation code.
* In main.cpp, organize includes and singletons.

### CODE REVIEW

**Comment by mhilbrunner** [comment]
PR review meeting: further internal cleanup may be nice, but not priority for now in 4.0 Beta phase.

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 1]
```diff
@@ -5767,8 +5766,8 @@ void CanvasItemEditorViewport::_create_nodes(Node *parent, Node *child, String &
 		undo_redo->add_undo_method(ed, "live_debug_remove_node", NodePath(String(EditorNode::get_singleton()->get_edited_scene()->get_path_to(parent)) + "/" + new_name));
 	}
 
-	if (Object::cast_to<TouchScreenButton>(child) || Object::cast_to<TextureButton>(child)) {
-		undo_redo->add_do_property(child, "texture_normal", texture);
+	if (child->has_method("set_texture_normal")) {
+		undo_redo->add_do_method(child, "set_texture_normal", texture);
```
Comment (akien-mga) [inline editor/plugins/canvas_item_editor_plugin.cpp:5770]: What's the rationale for this change?

**Review by akien-mga — APPROVED** [review]
Looks fine overall.

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 2]
```diff
@@ -4689,7 +4689,6 @@ void CanvasItemEditor::_popup_callback(int p_op) {
 				undo_redo->add_undo_method(this, "_set_owner_for_node_and_children", n2d, editor_root);
 			}
 			undo_redo->commit_action();
-
```
Comment (AThousandShips) [inline editor/plugins/canvas_item_editor_plugin.cpp:4692]: This space is used in the other cases, I'd restore it for uniformity

**Review by AThousandShips — APPROVED** [review]
Looks good to me

File: editor/plugins/canvas_item_editor_plugin.cpp (hunk 1 above)
Comment (aaronfranke) [reply editor/plugins/canvas_item_editor_plugin.cpp:5770]: Original motivation: This avoids a hard dependency on `TouchScreenButton` which is in fact not a `Button` in Godot but instead inherits `Node2D`, so it can't be used if compiling without `Node2D`.

Intent: Instead of looking for classes that have `texture_normal`, let's just look for `texture_normal`.

However, since there is no `has_property`, I had to use `has_method` instead. We should consider adding `has_property` though, that would be better I think.

Comment (akien-mga) [reply editor/plugins/canvas_item_editor_plugin.cpp:5770]: There's another hard dependency on `TouchScreenButton` in this class though, so it doesn't really solve anything as is:
```cpp
        // Adjust position for Control and TouchScreenButton
        if (Object::cast_to<Control>(child) || Object::cast_to<TouchScreenButton>(child)) {
                target_position -= texture->get_size() / 2;
        }
```

Comment (aaronfranke) [reply editor/plugins/canvas_item_editor_plugin.cpp]: True, this is non-trivial. I'll remove for this PR.

**Comment by akien-mga** [comment]
Thanks!
